integration_tests/src/main/python/join_test.py

```diff
@@ -78,7 +105,7 @@ def do_join(spark):
                          ('b', StringGen()), ('c', BooleanGen())]
 
 @ignore_order
-@pytest.mark.parametrize('join_type', ['Inner', 'LeftSemi', 'LeftAnti'], ids=idfn)
+@pytest.mark.parametrize('join_type', ['Left', 'Inner', 'LeftSemi', 'LeftAnti', 'FullOuter'], ids=idfn)
```

Comment:
Curious, this looks like it does an outer join with nulls but isn't marked xfail as `test_broadcast_join` is?

Reply:
Ah nevermind, I now see this version is joining on a null-safe equals.

Reply:
Spark is smart enough that it sees the comparison operator and knows it can removes the nulls early in the process.